includes/Classifai/Admin/SavePostHandler.php

```diff
@@ -21,26 +21,27 @@ public function register() {
 	}
 
 	/**
-	 * Save Post handler only runs on admin or REST requests
+	 * Check to see if we can register this class.
 	 */
 	public function can_register() {
-		if ( ! get_option( 'classifai_configured', false ) ) {
-			return false;
-		} elseif ( empty( get_option( 'classifai_watson_nlu' ) ) ) {
-			return false;
-		} elseif ( empty( get_option( 'classifai_watson_nlu' )['credentials']['watson_url'] ) ) {
-			return false;
-		} elseif ( is_admin() ) {
-			return true;
-		} elseif ( $this->is_rest_route() ) {
-			return true;
-		} elseif ( defined( 'PHPUNIT_RUNNER' ) && PHPUNIT_RUNNER ) {
-			return false;
-		} elseif ( defined( 'WP_CLI' ) && WP_CLI ) {
-			return false;
-		} else {
-			return false;
+
+		$should_register = false;
+		if ( is_admin() || $this->is_rest_route() ) {
```

Comment:
I think we need to add more conditions here. Right now, this class will be registered any time we're in the admin, even if things haven't been configured yet. I think we need to keep the configuration checks (the `classifai_configured` and `classifai_watson_nlu` options)

Reply:
Yep, that is way more clean, I like it!

Reply:
@dkotter I've pushed the requested changes, and updated the unit tests.